packages/common/src/BucketsConfig.ts

```diff
@@ -1,22 +1,24 @@
-'use strict';
+import { createErrorType } from '@cumulus/errors';
+import { BucketsConfigObject, BucketType } from './types';
 
-const { createErrorType } = require('@cumulus/errors');
 const BucketsConfigError = createErrorType('BucketsConfigError');
 
 /**
  * Class representing cumulus bucket configuration.
  */
 class BucketsConfig {
-  constructor(bucketsJsonObject) {
+  private readonly buckets: BucketsConfigObject;
+
+  constructor(bucketsJsonObject: BucketsConfigObject) {
     this.buckets = bucketsJsonObject;
   }
 
   /**
-   * returns key into this.buckets who's object has `name` bucketName
+   * returns key into this.buckets whose object has `name` bucketName
    * @param {string} bucketName
-   * @returns {string} desired bucket's key value.
+   * @returns {string|undefined} desired bucket's key value.
    */
-  key(bucketName) {
+  key(bucketName: string) {
```

Comment:
Do we want to explicitly type the return of these methods (and functions in general)?

Like:

```ts
key(bucketName: string): string | undefined {
```

Reply:
I'll try to find a link in the docs that talks about this, but recommended best practice that I've seen in a few places is to use TypeScript's automatically-determined return values. The only time that you want to override the automatically-generated one is in the few cases where TS _can't_ properly determine the return type. An example of that would be something like `return get(something, 'blah.woo');`, where TS isn't going to be able to automatically determine what the return value is.